### ISSUE TITLE
configuration.h

### ISSUE THREAD

**Issue by aryabhatt** [AdvancedPhotonSource/xpcs-eigen#6]
Hi,

Can you please consider moving the "Configuration::init" out of configuration.cpp to a separate file?
Also rename it to something more explicit, for example, "Configuration::from_hdf5". 

This will allow us to configure from multiple options, e.g. python object, json etc.

**Comment by iamfaisalkhan** [comment]
Yes sure that is fine by me. Let me know if you want to send a pull request for this. I will also go ahead and merge the current one tomorrow. Thank you!
